src/spclient.rs

```diff
@@ -359,13 +457,17 @@ impl SpClient {
         Ok(())
     }
 
-    pub fn set_fees(psbt: &mut Psbt, fee_rate: u32, payer: String) -> Result<()> {
-        let payer_vouts: Vec<u32> = match SilentPaymentAddress::try_from(payer.clone()) {
+    pub fn set_fees(psbt: &mut Psbt, fee_rate: Amount, payer: String) -> Result<()> {
```

Comment:
`fee_rate` is set as `Amount` here, but technically it's `Amount/vbyte`. I think that's important because we probably want to allow for fractions in the fee rate (e.g. 1.5 sats/vbyte)

Reply:
Let's think about that later, operation with Float values are annoying